### PR TITLE
[sovereign] Create epoch start data cross chain

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Create and display last cross chain notarized data
  
## Proposed changes
- Create/add pre-genesis cross chain header
- Accept genesis header without pre-genesis header for nodes with disabled notifier or start in epoch

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/track/sovereignChainShardBlockTrack.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func (scsbt *sovereignChainShardBlockTrack) receivedExtendedShardHeader(
 	scsbt.blockProcessor.ProcessReceivedHeader(extendedShardHeaderHandler)
 }
 
-func (scsbt *sovereignChainShardBlockTrack) isGenesisLastCrossNotarizedHeader() bool {
+func (scsbt *sovereignChainShardBlockTrack) IsGenesisLastCrossNotarizedHeader() bool {
```
Comment (axenteoctavian) [inline process/track/sovereignChainShardBlockTrack.go:129]: Missing function comment

File: process/block/sovereignChainBlock.go [hunk 2]
```diff
@@ -829,14 +834,30 @@ func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() er
 		return err
 	}
 
-	log.Trace("checkExtendedShardHeadersValidity", "lastCrossNotarizedHeader nonce", lastCrossNotarizedHeader.GetNonce())
 	extendedShardHdrs := scbp.sortExtendedShardHeadersForCurrentBlockByNonce()
 	if len(extendedShardHdrs) == 0 {
 		return nil
 	}
 
+	log.Trace("checkExtendedShardHeadersValidity",
```
Comment (axenteoctavian) [inline process/block/sovereignChainBlock.go:842]: Did you move this logging here with intention?

File: process/block/sovereignChainBlock.go [hunk 3]
```diff
@@ -829,14 +834,30 @@ func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() er
...
+		"lastCrossNotarizedHeader nonce", lastCrossNotarizedHeader.GetNonce(),
+		"lastCrossNotarizedHeader round", lastCrossNotarizedHeader.GetRound(),
+	)
+	if scbp.receivedGenesisMainChainHeaderWithoutPreGenesis(extendedShardHdrs[0]) {
+		// we are missing pre-genesis header, so we can't link it to previous header
+		if len(extendedShardHdrs) == 1 {
+			return nil
+		}
+
+		lastCrossNotarizedHeader = extendedShardHdrs[0]
```
Comment (axenteoctavian) [inline process/block/sovereignChainBlock.go:852]: The `extendedShardHdrs[0]` is not logged

File: process/track/sovereignChainShardBlockTrack.go (hunk 1 above)
Comment (mariusmihaic) [reply process/track/sovereignChainShardBlockTrack.go:129]: Right, added it

File: process/block/sovereignChainBlock.go (hunk 3 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:852]: Right, we could log it as well.

File: process/block/sovereignChainBlock.go [hunk 4]
```diff
@@ -848,6 +869,14 @@ func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() er
 	return nil
 }
 
+// this will return true if we receive the genesis main chain header in following cases:
+//   - no notifier is attached => we did not track main chain and don't have pre-genesis header
+//   - node is in re-sync/start in the exact epoch when we start to notarize main chain => no previous
+//     main chain tracking(notifier is also disabled)
+func (scbp *sovereignChainBlockProcessor) receivedGenesisMainChainHeaderWithoutPreGenesis(incomingHeader data.HeaderHandler) bool {
```
Comment (sasurobert) [inline process/block/sovereignChainBlock.go:882]: this function name is really confusing. Can you find a better name?

**Review by sasurobert — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

File: process/block/sovereignChainBlock.go (hunk 4 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go]: Right, good point, changed it

**Pull request merged**